src/builder-manifest.c

```diff
@@ -3754,41 +3754,25 @@ builder_manifest_show_deps (BuilderManifest *self,
 }
 
 static gboolean
-builder_manifest_install_dep (BuilderManifest *self,
-                              BuilderContext  *context,
-                              const char *remote,
-                              gboolean opt_user,
-                              const char *opt_installation,
-                              const char *runtime,
-                              const char *version,
-                              gboolean opt_yes,
-                              GError         **error)
+builder_manifest_install_single_dep (const char *ref,
+				     const char *remote,
+				     gboolean opt_user,
+				     const char *opt_installation,
+				     GError **error)
 {
-  g_autofree char *ref = NULL;
-  g_autofree char *commit = NULL;
   g_autoptr(GPtrArray) args = NULL;
-  gboolean installed;
-
-  if (version == NULL)
-    version = builder_manifest_get_runtime_version (self);
-
-  ref = flatpak_build_untyped_ref (runtime,
-                                   version,
-                                   builder_context_get_arch (context));
-
-  commit = flatpak_info (opt_user, opt_installation, "--show-commit", ref, NULL);
-  installed = (commit != NULL);
-
   args = g_ptr_array_new_with_free_func (g_free);
   g_ptr_array_add (args, g_strdup ("flatpak"));
   add_installation_args (args, opt_user, opt_installation);
-  if (installed)
+  if (remote == NULL)
     {
+      g_print("Updating %s\n", ref);
       g_ptr_array_add (args, g_strdup ("update"));
       g_ptr_array_add (args, g_strdup ("--subpath="));
     }
   else
     {
+      g_print("Trying to install %s from %s\n", ref, remote);
```

Comment:
I think we should move this print to builder_manifest_install_dep() and skip the "trying" prefix if there is only one remote.
So, it'll be like for the standard case the output would look something like:
```
Installing $ref
... output, including an error
There was an error installing the dependency $dep
exit(1)
```

But for the multi-remote case:
```
Trying to installing $ref from $remote1
... output, including an error
Trying to installing $ref from $remote2
... output, including an error
There was an error installing the dependency $dep
exit(1)
```

Reply:
More or less like you described now.
```
flatpak-builder --user --install-deps-from=flathub --repo=steam --force-clean build-dir com.valvesoftware.Steam.yml
Dependency Sdk: org.freedesktop.Sdk 19.08
Installing org.freedesktop.Sdk/x86_64/19.08 from flathub
error: Nothing matches org.freedesktop.Sdk in remote flathub
Error installing deps: running `flatpak --user install flathub -y --noninteractive org.freedesktop.Sdk/x86_64/19.08`: Child process exited with code 1
```

```
flatpak-builder --user --install-deps-from flathub --install-deps-from=flathub --repo=steam --force-clean build-dir com.valvesoftware.Steam.yml
Dependency Sdk: org.freedesktop.Sdk 19.08
Trying to install org.freedesktop.Sdk/x86_64/19.08 from flathub
error: Nothing matches org.freedesktop.Sdk in remote flathub
Trying to install org.freedesktop.Sdk/x86_64/19.08 from flathub
error: Nothing matches org.freedesktop.Sdk in remote flathub
Error installing deps: running `flatpak --user install flathub -y --noninteractive org.freedesktop.Sdk/x86_64/19.08`: Child process exited with code 1
```

Reply:
Hmm, looks like I misread your suggestion and now even the single-remote case prints out the remote. Is this fine?